src/app/InteractionModelEngine.cpp

```diff
@@ -95,20 +96,9 @@ void InteractionModelEngine::Shutdown()
 {
     mpExchangeMgr->GetSessionManager()->SystemLayer()->CancelTimer(ResumeSubscriptionsTimerCallback, this);
 
-    CommandHandlerInterface * handlerIter = mCommandHandlerList;
-
-    //
-    // Walk our list of command handlers and de-register them, before finally
-    // nulling out the list entirely.
-    //
-    while (handlerIter)
-    {
-        CommandHandlerInterface * nextHandler = handlerIter->GetNext();
-        handlerIter->SetNext(nullptr);
-        handlerIter = nextHandler;
-    }
-
-    mCommandHandlerList = nullptr;
+    // TODO: individual object clears the entire command handler interface registry.
+    //       This may not be expected.
```

Comment:
Right, because the scoping of the registry is a bit weird; see above.

Reply:
Will create a singleton and will remove this clear in that PR.